lib/SIL/SILVerifier.cpp

```diff
@@ -1706,10 +1706,20 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     if (UI->getElement()->hasAssociatedValues()) {
       require(UI->getOperand()->getType().isObject(),
               "EnumInst operand must be an object");
-      SILType caseTy = UI->getType().getEnumElementType(UI->getElement(),
-                                                        F.getModule());
-      require(caseTy == UI->getOperand()->getType(),
-              "EnumInst operand type does not match type of case");
+      // FIXME: Relax this check to look more like ApplyInst argument type
+      // verification.  It's possible that we emit an uncurried EnumInst
+      // payload sans argument labels:
+      //
+      // enum Expr {
+      //   indirect case elet(locals: [(String, Expr)], body: Expr)
+      // }
+      //
+      // Expr.elet(locals: body:)([], someExpr)
```

Comment:
@gottesmm Any thoughts here?

Reply:
What is "an uncurried EnumInst payload sans argument labels"?

Reply:
I.e., can you elaborate?

Reply:
See the example in the comment.  The old check does strict type equality (labels included) rather than element-by-element equality (labels excluded)

Reply:
Can you show me what the SIL looks like?

Reply:
[Sure](https://gist.github.com/CodaFi/e370b0b25d6bd5cb81ec36b21daf1896#file-elet-sil-L50)

Reply:
(I can't really say anything about SIL verification without seeing the SIL).

If you had said that sometimes you pass the payload boxed to the enum, I would have understood you immediately.

I would say the check should be that it is either the payload type or a boxed payload type.

Reply:
(i.e. we had a language issue = p).